copycat/api/src/main/java/org/apache/kafka/copycat/data/ISchema.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.copycat.data;
+
+import java.util.List;
+
+/**
+ * @see Schema
+ * @see SchemaBuilder
+ */
+public interface ISchema {
```

Comment:
I'd actually prefer not to have this interface at all, if possible. The reason it exists right now is that it's nice to be able to validate values against the SchemaBuilder while still building a schema because you need to do that to check default values. Without this you either need to a) implement `validateValue`, with basically identical code, in both SchemaBuilder (for default values) and Schema (in order to check that an object actually matches the schema), b) pay for the cost of constructing an intermediate Schema object to validate it (which in this implementation also requires building a hashmap of fields if it is struct), or c) accept that the validation will be disconnected from the `defaultValue()` method call and shift the check to occur when you call `build()`.